src/RegistryCoordinator.sol

```diff
@@ -758,33 +758,53 @@ contract RegistryCoordinator is
     }
 
-    /// @notice Returns the indices of the quorumBitmaps for the provided `operatorIds` at the given `blockNumber`
+    /**
+     * @notice Returns the indices of the quorumBitmaps for the provided `operatorIds` at the given `blockNumber`
+     * @dev Reverts if any of the `operatorIds` was not (yet) registered at `blockNumber`
+     * @dev This function is designed to find proper inputs to the `getQuorumBitmapAtBlockNumberByIndex` function
+     */
     function getQuorumBitmapIndicesAtBlockNumber(
         uint32 blockNumber, 
         bytes32[] memory operatorIds
     ) external view returns (uint32[] memory) {
         uint32[] memory indices = new uint32[](operatorIds.length);
         for (uint256 i = 0; i < operatorIds.length; i++) {
-            uint256 length = _operatorBitmapHistory[operatorIds[i]].length;
-            for (uint256 j = 0; j < length; j++) {
-                if (_operatorBitmapHistory[operatorIds[i]][length - j - 1].updateBlockNumber <= blockNumber) {
-                    uint32 nextUpdateBlockNumber = 
-                        _operatorBitmapHistory[operatorIds[i]][length - j - 1].nextUpdateBlockNumber;
-                    require(
-                        nextUpdateBlockNumber == 0 || nextUpdateBlockNumber > blockNumber,
-                        "RegistryCoordinator.getQuorumBitmapIndicesAtBlockNumber: operatorId has no quorumBitmaps at blockNumber"
-                    );
-                    indices[i] = uint32(length - j - 1);
-                    break;
-                }
-            }
+            indices[i] = _getQuorumBitmapIndexAtBlockNumber(blockNumber, operatorIds[i]);
         }
         return indices;
     }
 
     /**
-     * @notice Returns the quorum bitmap for the given `operatorId` at the given `blockNumber` via the `index`
-     * @dev reverts if `index` is incorrect 
+     * @notice Returns the index of the quorumBitmap for the provided `operatorId` at the given `blockNumber`
+     * @dev Reverts if the operator had not yet (ever) registered at `blockNumber`
+     * @dev This function is designed to find proper inputs to the `getQuorumBitmapAtBlockNumberByIndex` function
+     */
+    function _getQuorumBitmapIndexAtBlockNumber(
+        uint32 blockNumber, 
+        bytes32 operatorId
+    ) internal view returns (uint32 index) {
+        uint256 length = _operatorBitmapHistory[operatorId].length;
+        for (uint256 i = 0; i < length; i++) {
+            if (_operatorBitmapHistory[operatorId][length - i - 1].updateBlockNumber <= blockNumber) {
+                uint32 nextUpdateBlockNumber = 
+                    _operatorBitmapHistory[operatorId][length - i - 1].nextUpdateBlockNumber;
+                require(
+                    nextUpdateBlockNumber == 0 || nextUpdateBlockNumber > blockNumber,
+                    "RegistryCoordinator.getQuorumBitmapIndexAtBlockNumber: operatorId has no bitmap update at blockNumber"
+                );
```

Comment:
I'm less a fan of leaving this in, for a few reasons. 

First, if we hit this, it means the contract's storage is corrupted and there's a bug outside this method.

IMO contracts should keep assertions like this to a minimum, primarily to cover important state changes. This is a getter used by offchain code, so I feel like it only confuses (I mean, we know this code and we had a hard time figuring out why it was here)

Second - if we really want to keep it, let's at least change the error message - it's wildly inaccurate and will lead to further confusion. Maybe something like `RegCoord.methodName: history is corrupted`?

Third - this method is really clean and straightforward without these extra lines of code. I like that, because currently the "state history" model our code uses is really confusing, and being able to look at the getters our offchain code is consuming makes things make more sense - unless they're bogged down with stuff like this

Reply:
fair enough. I pointed out that it could be removed to elicit some feedback. I don't feel strongly either way, but you've convinced me to remove it.

Reply:
code removed here https://github.com/Layr-Labs/eigenlayer-middleware/pull/121/commits/8cd21f4d8754453d5f90948b9be74fe2885d8230